core/src/main/java/org/apache/accumulo/core/metadata/schema/TabletsMetadata.java

```diff
@@ -290,7 +290,8 @@ public TableRangeOptions forTable(TableId tableId) {
     @Override
     public Options forTablet(KeyExtent extent) {
       forTable(extent.tableId());
-      this.range = new Range(extent.toMetaRow());
+      // this.range = new Range(extent.toMetaRow());
+      this.range = new Range(extent.toMetaRange());
```

Comment:
`extent.toMetaRange()` already returns a `Range` type. It's not necessary to copy it again.

```suggestion
      this.range = extent.toMetaRange();
```

This change to use the `toMetaRange` rather than `toMetaRow` seems risky. There are many places in our code where we tend to throw "null" into the prevEndRow when constructing a KeyExtent, just because we know we were using it only to carry the TableId and endRow, and we didn't necessarily care about the value of the prevEndRow. Since this change now makes us actually use the prevEndRow, we need to make sure that any callers aren't breaking things by using null for the prevEndRow when constructing the KeyExtent that is used here.

I don't have a good grasp of the assumptions that code that calls this might make, and how it might be affected by this change. This probably warrants further review.

Reply:
I agree, this option is less straight forward. If it seems like the other change will suffice I will remove this one. Either way I'll clean it up with your suggestion for now.

Reply:
One thing to do is check to see where all we are constructing a KeyExtent with a null prevEndRow, to see if they would be affected by these changes.

Reply:
I think it would be best to stick with `new Range(extent.toMetaRow());` for this code.  That creates a range that will return 0 or 1 tablets.   Using `extent.toMetaRange()` create a range that will return 0 or more tablets.  For this case we want to get a single tablet, if it exists otherwise nothing.